lib/datadog/tracing/span.rb

```diff
@@ -136,7 +140,8 @@ def to_hash
           service: @service,
           span_id: @id,
           trace_id: @trace_id,
-          type: @type
+          type: @type,
+          span_links: @links.map!(&:to_hash)
```

Comment:
This is modifying the original `@links` list of objects, which you probably don't want. Use `map` instead.